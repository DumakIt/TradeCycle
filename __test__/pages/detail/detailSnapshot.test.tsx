import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { RecoilRoot } from "recoil";
import DetailPage from "../../../pages/[detail]";
import {
  ApolloClient,
  ApolloProvider,
  HttpLink,
  InMemoryCache,
} from "@apollo/client";
import fetch from "cross-fetch";

jest.mock("next/router", () => require("next-router-mock"));

it("detail 페이지 snapshot 테스트", () => {
  const client = new ApolloClient({
    link: new HttpLink({
      uri: "http://mock.com/graphql",
      fetch,
    }),
    cache: new InMemoryCache(),
  });

  const result = render(
    <RecoilRoot>
      <ApolloProvider client={client}>
        <DetailPage />
      </ApolloProvider>
    </RecoilRoot>
  );
  expect(result.container).toMatchSnapshot();
});
