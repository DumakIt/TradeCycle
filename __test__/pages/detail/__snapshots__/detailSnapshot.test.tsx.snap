// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail 페이지 snapshot 테스트 1`] = `
<div>
  <div
    class="css-lu91pw"
  >
    <div
      class="css-1rix88y"
    >
      <img
        src="https://storage.googleapis.com/"
      />
    </div>
    <div
      class="css-1d72qcm"
    >
      <div
        class="css-1aku9oj"
      >
        <span
          aria-label="edit"
          class="anticon anticon-edit css-eg37wo"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
            />
          </svg>
        </span>
        <span
          aria-label="close"
          class="anticon anticon-close css-h3gfdb"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="close"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </span>
      </div>
      <div
        class="css-foivsi"
      />
      <div
        class="css-fxouc"
      >
        <span>
          원
        </span>
      </div>
      <div
        class="css-1omltoe"
      >
        <button>
          구매하기
        </button>
      </div>
    </div>
  </div>
  <div
    class="css-156j2qy"
  >
    <div
      class="css-g11p7y"
    >
      <div
        class="css-7m9ncl"
      >
        상품정보
      </div>
      <div
        class="css-hc3rel"
      />
      <div
        class="css-1p6kesl"
      />
      <div
        class="css-19rft6q"
      />
      <div
        class="css-9gqlgs"
      />
    </div>
    <div
      class="css-16z20vs"
    />
    <div
      class="css-pl1dzu"
    >
      <div
        class="css-7m9ncl"
      >
        판매자
      </div>
      <div
        class="css-c417b1"
      >
        <div
          class="css-637fsy"
        >
          <img
            src="https://storage.googleapis.com/"
          />
        </div>
        <div
          class="css-1m0t0nf"
        />
      </div>
      <div
        class="css-ot0n1b"
      >
        댓글
      </div>
      <div
        class="css-ee0hcs"
      />
      <form>
        <div
          class="css-1eephp8"
        >
          <textarea
            name="contents"
          />
          <button
            class="css-1an7oxl"
          >
            작성하기
          </button>
        </div>
      </form>
      <div
        class="css-12kcw7f"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
